src/wasm/wasm-stack.cpp

```diff
@@ -65,6 +65,60 @@ void BinaryInstWriter::visitLoop(Loop* curr) {
 void BinaryInstWriter::visitBreak(Break* curr) {
   o << int8_t(curr->condition ? BinaryConsts::BrIf : BinaryConsts::Br)
     << U32LEB(getBreakIndex(curr->name));
+
+  // See comment on |brIfsNeedingHandling| for the extra casts we need to emit
+  // here for certain br_ifs.
+  auto iter = brIfsNeedingHandling.find(curr);
+  if (iter != brIfsNeedingHandling.end()) {
+    auto unrefinedType = iter->second;
+    auto type = curr->type;
+    assert(type.size() == unrefinedType.size());
+
+    assert(curr->type.hasRef());
+
+    auto emitCast = [&](Type to, Type from) {
```

Comment:
Since the stringview PR has now landed, there's no need to supply the `from` type anymore, so this can be simplified a bit. I guess we still need the `unrefinedType` in the `brIfsNeedingHandling` map for the scratch locals, though.

Reply:
Yeah, `from` can go away but we do still need to know the unrefined type because the scratch locals are of that type.